product_contract/models/sale_order.py

```diff
@@ -41,11 +41,20 @@ def _compute_is_contract(self):
     @api.multi
     def _prepare_contract_value(self, contract_template):
         self.ensure_one()
+        journal = self.env['account.journal'].search(
+            [
+                ('type', '=', contract_template.contract_type),
+                ('company_id', '=', self.company_id.id),
+            ],
+            limit=1,
+        )
```

Comment:
Would it make sense to use `contract_template.journal_id`? I'm a bit worried that searching journal types based on contract types would lead to bugs in the future, because these two types are independent in theory.

Reply:
It's the same logic used to get the default journal when creating a contract or contract template

Reply:
Ok. But is it not a good opportunity to let users specify the journal on the contract template?

Reply:
if the journal is set on the contract template, the result get by this search will be overridden by the onchange method called [here](https://github.com/OCA/contract/pull/422/files/ab177a2999156a5bca9c60b58330abcc30fdc897#diff-7e52e92beab9411dd73aa2b44d0945fdR93)

See also [this](https://github.com/OCA/contract/pull/422/files/ab177a2999156a5bca9c60b58330abcc30fdc897#diff-26c1ee0949150674b4450ea66bd9b8e9R213)